src/java/org/apache/cassandra/thrift/CustomTThreadPoolServer.java

```diff
@@ -198,6 +204,7 @@ public void run()
             TProtocol inputProtocol = null;
             TProtocol outputProtocol = null;
             SocketAddress socket = null;
+            ThriftSessionManager.instance.trackClient(client_);
```

Comment:
Instead of tracking all clients in a map, can we not just close the client socket "if stopped" in the `finally` ?

Reply:
the problem is you can still have clients being processed for minutes after shutting down the server socket, so closing the socket there would be too late